lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -25,533 +26,454 @@
 
 using namespace circt;
 using namespace firrtl;
-struct Node {
-  Value val = nullptr;
-  size_t fieldId = 0;
-  Node() = default;
-  Node(Value val, size_t fieldId) : val(val), fieldId(fieldId) {}
-  Node(const Node &rhs) = default;
-  Node &operator=(const Node &rhs) = default;
-  bool operator==(const Node &rhs) const {
-    return (val == rhs.val && fieldId == rhs.fieldId);
-  }
-  bool operator!=(const Node &rhs) const { return !(*this == rhs); }
-  bool isValid() { return val != nullptr; }
-  FModuleOp getParentModule() {
-    assert(val && "Must be a valid node");
-    if (val.isa<BlockArgument>())
-      return cast<FModuleOp>(val.getParentBlock()->getParentOp());
-    return val.getDefiningOp()->getParentOfType<FModuleOp>();
-  }
-};
 
-class ModuleDFS {
-  using ValueOpPair = std::pair<Value, Operation *>;
+using SetOfNodes = std::set<FieldRef>;
```

Comment:
`std::set` for stable iterators?

Reply:
Was not intentional, but updated to use `DenseSet`, throughout.